plugins/catalog-backend/src/service/CatalogPlugin.ts

```diff
@@ -61,6 +64,10 @@ class CatalogProcessingExtensionPointImpl
     this.#placeholderResolvers[key] = resolver;
   }
 
+  setEntityDataParser(parser: CatalogProcessorParser): void {
+    this.entityDataParser = parser;
```

Comment:
suggestion:
At a few other places, we throw an exception when such a setter is called twice (e.g., two competing modules). To 
achieve this, the properties are optional and the default is applied when using the value.

Reply:
Make sense to implement the same behaviour, I will look into this.